### PR TITLE
Port styletransfer code using generators

### PR DESCRIPTION
**Pull request by gabrielelanaro**
This is useful because generators make implementing streaming API easier.

### CODE REVIEW

File: prototypes/styletransfer/model.py [hunk 1]
```diff
@@ -116,15 +116,17 @@ def run_style_transfer(
...
             content_img, style_img
         )
         gram_style_features = [
             gram_matrix(style_feature) for style_feature in style_features
         ]
 
         # Set initial image
-        init_image = tfe.Variable(content_img, dtype=tf.float32)
+        # TODO(glanaro): is it correct to preprocess or we don't really care?
+        init_img = self._process_img(content_img)
```
Comment (FraPochetti) [inline prototypes/styletransfer/model.py]: Using `content_img` to initialize `init_img` does not work well in practice, in my experiments, as the `content` component of the loss is, by definition, counted twice. Starting from a random noise image yields more visually appealing results, with a stronger `style` component. We need to explore this further.

Comment (FraPochetti) [reply prototypes/styletransfer/model.py]:
```
def run_style_transfer(
        self,
        content_img: np.array,
        style_img: np.array,
        num_iterations=1000,
        content_weight=1e3,
        style_weight=1e-2,
    ):
```
`content_weight` and `style_weight` need to be calculated on the fly as they vary according to the image. We need to adjust the weights so that content and style don't show orders of magnitude of difference (which is the default case). To give an example, in my experiments, I was often obliged to set `content_weight=1000` and `style_weight=1`. Ideally we need to run a couple of optimization iterations and compute the content2style ratio, then use it to execute the actual style transfer process. As soon as the 2 losses are comparable, then we can add a custom coefficient to let the end user decide which proportion of content/style he desires.

File: prototypes/styletransfer/model.py [hunk 2]
```diff
@@ -116,15 +116,17 @@ def run_style_transfer(
...
+        content_features, style_features = self.feature_representations(
             content_img, style_img
         )
         gram_style_features = [
             gram_matrix(style_feature) for style_feature in style_features
         ]
 
         # Set initial image
-        init_image = tfe.Variable(content_img, dtype=tf.float32)
+        # TODO(glanaro): is it correct to preprocess or we don't really care?
```
Comment (FraPochetti) [inline prototypes/styletransfer/model.py]: I think we must, as `init_img`, `content_img` and `style_img` pass through the same NN and we compare the activations of the intermediate layers in the loss function. Pre-processing (ImageNet stats normalization) is of paramount importance.

Comment (gabrielelanaro) [reply prototypes/styletransfer/model.py]: Great, I'll remove the TODO

File: prototypes/styletransfer/model.py (hunk 1 above)
Comment (gabrielelanaro) [reply prototypes/styletransfer/model.py]: For the content2style ratio I created a new issue because it's outside of the scope of this PR, for the image initialization I'll add both versions (random or content) as a configuration option.

File: prototypes/styletransfer/model.py (hunk 2 above)
Comment (FraPochetti) [reply prototypes/styletransfer/model.py]: By accident, I once applied custom normalization (on the validation set) instead of using ImageNet stats  (which I was using on the training set) and the end result on the Deep Learning task was disastrous. Training accuracy dropping to 60% from 95% (binary classification). I could not believe it had such a huge impact. The truth is that the network "sees" a completely different image if preprocessed improperly.

**Comment by gabrielelanaro** [comment]
> Code-wise we are good. I added a couple of ML-related comments which we can investigate offline on gitter.
> 
> **An additional remark**: how are we supposed to ingest real images into this pipeline?
> It is not immediate to me how to read an image from disk and pass it to the routine.
> 
> **EDIT**: I guess everything starts with `.images.load_image(path_to_image)`, which returns a `np.array`, right?

Real images can come from a filesystem (load_png) or whatever, or as numpy images, that is correct.
